ipa-core/src/protocol/ipa_prf/malicious_security/prover.rs

```diff
@@ -296,7 +450,16 @@ mod test {
         const U: [u128; 1024] = [1u128; 1024];
         const V: [u128; 1024] = [2u128; 1024];
 
-        let uv_before = zip_chunks(&U, &V);
+        let uv_before = U
+            .iter()
+            .zip(V.iter())
+            .map(|(u, v)| {
+                (
+                    Fp61BitPrime::truncate_from(*u),
+                    Fp61BitPrime::truncate_from(*v),
+                )
+            })
+            .collect::<UVValues<Fp61BitPrime, 32>>();
```

Comment:
See how this code is duplicated with the code above on lines 411-420? When you have that sort of thing going on, it's a good indication that we should wrap this in an abstraction - like I previously did with "zip_chunks()". Maybe make a function to do this called "gen_uv_store()". In fact, this is serving a nearly identical purpose to "zip_chunks()", so maybe you should just eliminate the "zip_chunks()" method entirely. Then on line 378 where you are just calling the current "zip_chunks()" you could replace it with "gen_uv_store().iter()"